openerp/addons/test_new_api/tests/test_cache.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+from openerp.tests.common import TransactionCase
+
+
```

Comment:
You may combine all tests in a single class, and test each case as a test method, like:
```python
class TestOne2many(common.TransactionCase):
    def check(self, multi):
        # check the lines first
        self.assertItemsEqual(multi.lines.mapped('name'), map(str, range(10)))
        # modify the first line and drop the last one
        multi.lines[0].name = "hello"
        multi.lines = multi.lines[:-1]
        self.assertEqual(len(multi.lines), 9)
        self.assertIn("hello", multi.lines.mapped('name'))
        # invalidate the cache and check again; this crashes if the value of
        # multi.lines in cache contains new records
        multi.invalidate_cache()
        self.assertEqual(len(multi.lines), 9)
        self.assertIn("hello", multi.lines.mapped('name'))

    def test_new(self):
        """ Check lines created with method new(). """
        multi = self.env['test_new_api.multi'].create({'name': "What's up?"})
        multi.lines = ...
        self.check(multi)

    def test_create(self):
        """ Check lines created with method create(). """
        multi = self.env['test_new_api.multi'].create({'name': "What's up?"})
        multi.lines = ...
        self.check(multi)

    def test_commands(self):
        """ Check lines created with commands. """
        multi = self.env['test_new_api.multi'].create({'name': "What's up?"})
        multi.lines = ...
        self.check(multi)
```